### PR TITLE
Feature/jake/add free text and enter key action params

### PR DESCRIPTION
**Pull request by jacoblipson**
- extend directive to accept and utilize new parameters:
  - freeText allows for the selection of items not provided in the dropdown
  - enterAction receives a function that executes whenever the enter key is pressed
- start a README for the library

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,63 @@
 # [ng-suggestion](http://ianwalter.github.io/ng-suggestion/)
```
Comment (teddywing) [inline README.md]: Can we update the URL to use `http://flashnotes.github.io/ng-suggestion/`?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,63 @@
...
 *Flexible AngularJS typeahead / autocomplete / suggestion / predictive search directive*
 
-I don't have time for docs right now just read the source.
-
+A wise man once said, *"I don't have time for docs right now just read the source."*
```
Comment (teddywing) [inline README.md]: Nice, but this should be removed when we get the README to a more complete state

File: README.md [hunk 3]
```diff
@@ -1,6 +1,63 @@
...
+bower install ng-suggestion --save
+```
+
+Include js/ng-suggestion.min.js in your app. You may also want to include
+css/ng-suggestion.min.css, but it's not necessary, only recommended.
+
+#### Setup for ng-suggestion
+
+```javascript
+var app = angular.module('your-wonderful-app', ['ng-suggestion'])
```
Comment (teddywing) [inline README.md]: Let's follow Angular's style conventions for naming modules. See for example those defined here: https://docs.angularjs.org/guide/module.

File: README.md [hunk 4]
```diff
@@ -1,6 +1,63 @@
...
+```
+
+#### Using ng-suggestion
+```html
+<input dropdown-field
+       suggestion="courseService.courseLookup"
+       suggestion-model="courseService.current.courseName"
+       suggestion-dropdown="courseService.courseLookupDropdown"
+       suggestion-url="courseService.courseLookupUrl"
+{#       suggestion-delete-handler="courseService.deleteNewSubject()"#}
```
Comment (teddywing) [inline README.md]: Why are there Django comment markers here?

File: README.md [hunk 5]
```diff
@@ -1,6 +1,63 @@
...
+       suggestion-params="{
+         q: courseService.current.courseName,
+         subject_id: courseService.current.subject.id
+       }"/>
```
Comment (teddywing) [inline README.md]: Do you mind adding a space here (`}" />`) to give this a little more breathing room?

File: README.md [hunk 6]
```diff
@@ -1,6 +1,63 @@
...
+```suggestion-free-text``` (optional) Allows the selection of items that are not in the
+populated list of options. The value is a boolean; the default behavior is not to allow.
+
+```suggestion-enter-action``` (optional) Executes a function when the enter key is hit.
+The value must be a function.
+
+```suggestion-response-property``` Needs Info
+
+
+*This README was last updated on 10/5/2015 by [Jake Lipson](jacob@luvolearn.com)*
```
Comment (teddywing) [inline README.md]: This line seems irrelevant to me since we can get all of this information from the commit history.

**Comment by jacoblipson** [comment]
Couple notes about the current state:
* src needs to be recompiled
* enterAction still does not function as intended

**Comment by jacoblipson** [comment]
Resolved the compiling issues (though still getting the errors in Terminal).

**Comment by jacoblipson** [comment]
@teddywing, having trouble installing my fork with bower to facilitate testing. When you're ready to take a look at this, let's talk?

File: src/js/ng-suggestion.js [hunk 7]
```diff
@@ -65,8 +65,14 @@
         this.keydownHandler = function(input) {
           return ($event) => {
             var currentOption = input.dropdown.currentOption;
-            if (currentOption && $event.keyCode === 13) { // Enter
+            var freeText = input.freeText;
+            var enterAction = input.enterAction;
```
Comment (teddywing) [inline src/js/ng-suggestion.js]: Are these extra variables necessary? Couldn't we use `input.freeText` and `input.enterAction` directly?

File: README.md [hunk 8]
```diff
@@ -1,6 +1,56 @@
-# [ng-suggestion](http://ianwalter.github.io/ng-suggestion/)
+# [ng-suggestion](https://github.com/flashnotes/ng-suggestion/)
```
Comment (teddywing) [inline README.md]: Let's make this the URL of the GH-pages demo page: `http://flashnotes.github.io/ng-suggestion/`

File: README.md [hunk 9]
```diff
@@ -1,6 +1,56 @@
...
-&mdash;&nbsp; [Ian Kennington Walter](http://iankwalter.com)
+#### Installing ng-suggestion
+
+Install using Bower:  
+```
+bower install ng-suggestion --save
+```
+
+Include js/ng-suggestion.min.js in your app. You may also want to include
+css/ng-suggestion.min.css, but it's not necessary, only recommended.
```
Comment (teddywing) [inline README.md:16]: Let's put this information in the "Setup" or "Using" sections. How about showing sample markup to include those files?

``` html
<link rel="stylesheet" media="screen" href="/bower_components/ng-suggestion/dist/css/ng-suggestion.min.css" />
```

``` html
<script src="/bower_components/ng-suggestion/dist/js/ng-suggestion.min.js"></script>
```

File: README.md [hunk 10]
```diff
@@ -1,6 +1,56 @@
...
+```suggestion-dropdown``` Needs Info
+
+```suggestion-url``` The URL of the API that will return the list of suggestions.
+
+```suggestion-delete-handler``` Needs Info
+
+```suggestion-params``` The parameters to be used to query the identified API.
+
+```suggestion-free-text``` (optional) Allows the selection of items that are not in the
+populated list of options. The value is a boolean; the default behavior is not to allow.
```
Comment (teddywing) [inline README.md]: Let's be explicit and say the default is `false`

**Comment by teddywing** [comment]
Looks good :rocket:
